contrib/qat/compression/qatzstd/compressor/test/qatzstd_compressor_impl_test.cc

```diff
@@ -0,0 +1,110 @@
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/stats/isolated_store_impl.h"
+#include "source/extensions/compression/zstd/decompressor/zstd_decompressor_impl.h"
+
+#include "test/mocks/server/factory_context.h"
+#include "test/test_common/utility.h"
+
+#include "contrib/qat/compression/qatzstd/compressor/source/config.h"
+#include "gtest/gtest.h"
+#include "qatseqprod.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Qatzstd {
+namespace Compressor {
+namespace {
+
+class QatzstdCompressorImplTest : public testing::Test {
+protected:
+  void drainBuffer(Buffer::OwnedImpl& buffer) {
+    buffer.drain(buffer.length());
+    ASSERT_EQ(0, buffer.length());
+  }
+
+  void verifyWithDecompressor(Envoy::Compression::Compressor::CompressorPtr compressor) {
+    Buffer::OwnedImpl buffer;
+    Buffer::OwnedImpl accumulation_buffer;
+    std::string original_text{};
+    for (uint64_t i = 0; i < 10; i++) {
+      TestUtility::feedBufferWithRandomCharacters(buffer, default_input_size_ * i, i);
```

Comment:
Does this always generate single slice buffer?

Reply:
Yeah, `feedBufferWithRandomCharacters` just generate single slice buffer.

Reply:
Yeah, `feedBufferWithRandomCharacters` just generate single slice buffer.

Reply:
It will be great we have some test can cover the case of multiple slices buffer

Reply:
Sure, done, thanks!